pinot-query-runtime/src/main/java/org/apache/pinot/query/service/server/QueryServer.java

```diff
@@ -81,7 +79,8 @@ public class QueryServer extends PinotQueryWorkerGrpc.PinotQueryWorkerImplBase {
   // query submission service is only used for plan submission for now.
   // TODO: with complex query submission logic we should allow asynchronous query submission return instead of
   //   directly return from submission response observer.
-  private final ExecutorService _querySubmissionExecutorService;
+  private final ExecutorService _submissionExecutorService;
+  private final ExecutorService _explainExecutorService;
```

Comment:
I decided to run explain in another thread pool, given that the work to be done in explain is a bit heavier than the one during submission